velox/expression/FunctionSignature.h

```diff
@@ -104,9 +104,18 @@ class TypeSignature {
     return baseName_ == rhs.baseName_ && parameters_ == rhs.parameters_;
   }
 
+  bool isConstant() const {
+    return isConstant_;
+  }
+
+  void setIsConstant(bool isConstant) {
```

Comment:
It would be nice to keep this class immutable. Since is-constant is not a property of the type, consider, adding a list of boolean is-constant flags to the FunctionSignature class.

```
  const TypeSignature returnType_;
  const std::vector<TypeSignature> argumentTypes_;
  const std::vector<bool> constantArguments_;
  const bool variableArity_;
```

Reply:
Fixed